lib/HLSL/DxilGenerationPass.cpp

```diff
@@ -1820,6 +1869,42 @@ void DxilGenerationPass::TranslateLocalDxilResourceUses(Function *F, std::unorde
   }
 }
 
+void DxilGenerationPass::RemoveLocalDxilResourceAllocas(Function *F) {
+  BasicBlock &BB = F->getEntryBlock(); // Get the entry node for the function
+  std::unordered_set<AllocaInst *> localResources;
+  for (BasicBlock::iterator I = BB.begin(), E = --BB.end(); I != E; ++I)
+    if (AllocaInst *AI = dyn_cast<AllocaInst>(I)) { // Is it an alloca?
+
+      bool isResource = HLModule::IsHLSLObjectType(AI->getAllocatedType());
+
+      if (ArrayType *AT = dyn_cast<ArrayType>(AI->getAllocatedType())) {
+        Type *EltTy = AT->getElementType();
+        while (isa<ArrayType>(EltTy)) {
+          EltTy = EltTy->getArrayElementType();
+        }
+        isResource = HLModule::IsHLSLObjectType(EltTy);
+        // TODO: support local resource array.
+        DXASSERT(!isResource, "local resource array");
+      }
+      if (isResource) {
+        localResources.insert(AI);
+      }
+    }
+
+  SSAUpdater SSA;
+  SmallVector<Instruction *, 4> Insts;
+
+  for (AllocaInst *AI : localResources) {
+    // Build list of instructions to promote.
+    for (User *U : AI->users())
+      Insts.emplace_back(cast<Instruction>(U));
+
+    ResourceRemover(Insts, SSA).run(AI, Insts);
```

Comment:
Why do you need to pass Insts twice (in constructor() and run())? Is it not saved in the constructor?